To: vim_dev@googlegroups.com
Subject: Patch 7.3.044
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.044
Problem:    The preview window opened by the popup menu is larger than
	    specified with 'previewheight'. (Benjamin Haskell)
Solution:   Use 'previewheight' if it's set and smaller.
Files:	    src/popupmnu.c


*** ../vim-7.3.043/src/popupmnu.c	2010-08-15 21:57:25.000000000 +0200
--- src/popupmnu.c	2010-10-27 17:04:31.000000000 +0200
***************
*** 558,565 ****
  	    win_T	*curwin_save = curwin;
  	    int		res = OK;
  
! 	    /* Open a preview window.  3 lines by default. */
  	    g_do_tagpreview = 3;
  	    resized = prepare_tagpreview(FALSE);
  	    g_do_tagpreview = 0;
  
--- 558,568 ----
  	    win_T	*curwin_save = curwin;
  	    int		res = OK;
  
! 	    /* Open a preview window.  3 lines by default.  Prefer
! 	     * 'previewheight' if set and smaller. */
  	    g_do_tagpreview = 3;
+ 	    if (p_pvh > 0 && p_pvh < g_do_tagpreview)
+ 		g_do_tagpreview = p_pvh;
  	    resized = prepare_tagpreview(FALSE);
  	    g_do_tagpreview = 0;
  
*** ../vim-7.3.043/src/version.c	2010-10-27 16:49:41.000000000 +0200
--- src/version.c	2010-10-27 17:08:37.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     44,
  /**/

-- 
"It's so simple to be wise.  Just think of something stupid to say
and then don't say it."        -- Sam Levenson

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
